src/GoProControl.cpp

```diff
@@ -1666,19 +1705,38 @@ bool GoProControl::listenResponse()
       delay(5);
     }
   }
-  while (_wifi_client.available() > 0)
+
+  if (mediatimer)
   {
-    _response_buffer[index++] = _wifi_client.read();
+    delay(10); //Add delay. Without, response can be not complete for getmedia.
   }
-  _response_buffer[index] = '\0';
-  _wifi_client.stop();
 
-  if (_debug)
+  if (_wifi_client.available() < sizeof(_response_buffer)) // Add verification to not overflow the buffer
```

Comment:
Be careful with `sizeof` it is a very bad friend, see:
https://www.geeksforgeeks.org/sizeof-operator-c/
https://stackoverflow.com/questions/37538/how-do-i-determine-the-size-of-my-array-in-c

Reply:
I can change it with "1500" wich is the size of this buffer if you prefer !

Reply:
I have a more stable idea, let's create a define in the .h file with this number, something like
#define MAX_RESPONSE_LEN 1500
then we pass this to both this function and when we create the _response_buffer array

Reply:
ok ! will do this this afternoon ^^

Reply:
finally I did it !